lib/Ptero/Proxy/Workflow.pm

```diff
@@ -49,10 +49,30 @@ sub BUILDARGS {
 
 sub concrete_workflow {
     my $self = shift;
+    return $self->_concrete_workflow($self->workflow_skeleton, $self->workflow_executions);
+}
+
+sub _concrete_workflow {
+    my ($self, $skeleton_hashref, $executions_hashref) = @_;
+
+    my $concrete_workflow = Ptero::Concrete::Workflow->new($skeleton_hashref);
+    $concrete_workflow->register_components();
```

Comment:
We can move this call to `register_components` to the end of the new subroutine.